File: frontend/src/app/models/route-preview.ts
import { BusDTO } from "./bus";
import { CoordinateDTO } from "./coordinateDTO";

export interface RoutePreviewDTO {
    id: string;
    departureTime: string;
    destinationTime: string;
    travelTime: number;
    busNumbers: string[];
    coordinates: CoordinateDTO[]; 
  }

File: backend/DTOs/RoutePreview.cs
namespace PublicTransportNavigator.DTOs
{
    public class RoutePreview
    {
        public long Id { get; set; }
        public TimeSpan DepartureTime { get; set; }
        public TimeSpan DestinationTime { get; set; }
        public int TravelTime { get; set; }
        public List<long> BusNumbers { get; set; } = [];
        public List<Coordinate> Coordinates { get; set; } = [];
    }
}
